### PR TITLE
Clear front proxy headers after authentication is complete

### PR DESCRIPTION
**Pull request by enj**
This matches the logic we have for the Authorization header as well as the impersonation headers.

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader.go [hunk 1]
```diff
@@ -184,6 +174,23 @@ func (a *requestHeaderAuthRequestHandler) AuthenticateRequest(req *http.Request)
...
 
+func ClearAuthenticationHeaders(h http.Header, nameHeaders, groupHeaders, extraHeaderPrefixes StringSliceProvider) {
+	for _, headerName := range nameHeaders.Value() {
+		h.Del(headerName)
+	}
+	for _, headerName := range groupHeaders.Value() {
+		h.Del(headerName)
+	}
+	for _, prefix := range extraHeaderPrefixes.Value() {
+		prefix := strings.ToLower(prefix)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader.go:185]: ToLower allocates, I'd suggest a hasPrefixIgnoreCase helper that checks lengths, then calls strings.EqualFold on substring matching the length of the prefix check

Comment (enj) [reply staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader.go]: Done.

**Pull request merged**